rfcs/0005-mbi-search.md

```diff
@@ -0,0 +1,65 @@
+# RFC Proposal
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0005-mbi-search` 
+* Start Date: 2020-1-15
+* RFC PR: [](https://github.com/rust-lang/rfcs/pull/0000)
+* JIRA Ticket(s):
+    * [BLUEBUTTON-1516](https://jira.cms.gov/browse/BLUEBUTTON-1516)
+
+The addition of a search by Medicare Beneficiary Identifier (MBI) facility to the BFD's Patient search end-point.   
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+    * [Patient Search](#patient-search)
+    * [Include Identifiers](#include-identifiers)
+    * [Hash Algorithm](#hash-algorithm)
+    * [Synthetic Beneficiaries](#synthetic-beneficiaries)
+* [Prior Art](#prior-art)
+
+## Motivation
+[Motivation]: #motivation
+
+The Medicare Beneficiary Identifier (MBI) is replacing the Health Insurance Claim Identifier (HICN). See [New Medicare Card Project](https://www.cms.gov/Medicare/New-Medicare-Card/9-13-18-NMC-ODF-Slides.pptx) for some background on this change. This change to the BFD adds to BFD the same capabilities for MBIs as currently exist for HICNs. 
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+The proposed solution consists of a new identifier for the Patient search FHIR end-point, a new include-identifiers projection, and the same hashing algorithm that BFD uses for HICN.
+
+### Patient Search
+The `Patient` end-point supports searching for resources with a matching MBI. The CMS classifies the MBI as personally identifiable information (PII). The request must hash the MBI to prevent sensitive PII from entering access logs. A new MBI based identifier system is recognized: `https://bluebutton.cms.gov/resources/identifier/mbi-hash`. 
+
+An example request for a sythentic beneficiary `1S00A00AA00` is:
+```
+curl 'https://<bfd address>:443/v1/fhir/Patient?identifier=https%3A%2F%2Fbluebutton.cms.gov%2Fresources%2Fidentifier%2Fmbi-hash%7C37c37d08d239f7f1da60e949674c8e4b5bb2106077cb0671d3dfcbf510ec3248&_format=application%2Fjson%2Bfhir'
+```
+
+### Include Identifiers
+The returned patient resource includes the HICN and MBI identifiers if the request has the `IncludeIdentifiers` HTTP header set. BFD supports new values for the `IdentifierIncludes` header. 
+
+| value    | identifiers added | Comments                   |
+|----------|-------------------|----------------------------|
+| <empty>  |                   | Default value              |
+| true     | HICN and MBI      | MBI is added to the result |
+| hicn     | HICN              | New value supported        |
+| mbi      | MBI               | New value supported        |
+| hicn,mbi | HICN and MBI      | New value supported        |
+
+### Hash Algorithm
+For convenience, MBI hashing uses the existing HICN hashing algorithm and pepper.
```

Comment:
Worth including that it's using the same itterations as well.

Reply:
Roger on adding the iterations to the section. 

What do you have in mind for future work and hashing not scaling?

Reply:
The future work I was thinking of has more to do with the process of on-boarding new peering partners. For a partner to use the hash, bfd has to communicate a shared secret to each partner. As more partners join operations concerning the shared secret (pepper) get more complicated (e.g. rolling it)

I'm not sure if this RFC is the right place for the discussion but it might be.

Reply:
Ok. I understand the concern. I don't have any suggestions to improve the situation, so I'm going to leave out the issue.